components/net/http_loader.rs

```diff
@@ -129,7 +128,7 @@ impl NetworkConnector for NetworkHttpRequestFactory {
     type Stream = PooledStream<HttpsStream<SslStream<HttpStream>>>;
 
     fn connect(&self, host: &str, port: u16, scheme: &str) -> Result<Self::Stream, HttpError> {
-        self.connector.connect(&replace_host(host), port, scheme)
+        self.connector.connect(host, port, scheme)
```

Comment:
<s>This implementation exists only to do that `replace_host` call, now that it is gone you can pass `&self.connector` again to `HyperRequest::with_connector` in `NetworkHttpRequestFactory::create` and remove this impl altogether.</s>

Hah, you killed the whole wrapper in next commit, nice!